include/open62541/server_pubsub.h

```diff
@@ -923,6 +924,36 @@ UA_Server_addSecurityGroup(UA_Server *server, UA_NodeId securityGroupFolderNodeI
 UA_StatusCode UA_EXPORT UA_THREADSAFE
 UA_Server_removeSecurityGroup(UA_Server *server, const UA_NodeId securityGroup);
 
+/**
+ * @brief This user defined callback to notify the user about the current status of the periodic SKS Pull request.
+ * @param server instance used to make sks pull request
+ * @param sksClientStatus
+ */
+typedef void
+(*UA_Server_sksPullRequestCallback)(UA_Server *server, UA_StatusCode sksPullRequestStatus, UA_Boolean sksPullSuccess, void* userData);
+
+/**
+ * @brief Sets the SKS client config used to call the GetSecurityKeys Method on SKS and get the
```

Comment:
Document if this is just for a one-time SKS pull request or whether it will repeat automatically once a new key is required.

Is there a way to remove or modify the SKS client once set?

if there is just one SKS client per security group, then we could just use the security group ID to remove.

Reply:
> Is there a way to remove or modify the SKS client once set?

the sks client is an internal instance user will never has access to it. However, the user can update the client config using `setSksClient` API for the next iteration of key fetch. The key rollover mechanism will check if new keys are needed then it will call `getSecurityKeysAndStoreFetchedKeys` method, which will take care of the new client lifecycle. After each SKS Pull request iteration the sks client will be deleted by a delayed callback `sksClientCleanupCb`.

Reply:
Sounds good. Please add what you described to the public API documentation.
1-2 sentences ought to be good enough.